### ISSUE TITLE
Example Repo (wgpu boids port)

### ISSUE THREAD

**Issue by tombh** [EmbarkStudios/rust-gpu#799]
I'm still fairly new to both Rust and GPU programming, but very excited about this project and been following it for quite a while (I even submitted the first compute example PR a few months ago). But I'm not sure I'm exactly the target audience at the moment? There's a lot of prior knowledge that seems to be expected, I'm not sure how much that's intentional?

One thing that's surprised me is that the examples are not actually that simple in terms of setting up a new project. For instance I didn't realise that this repo can actually just be used as a Cargo dependency, even though the various crates aren't officially published yet. Based on that new-found knowledge I've made a standalone repo that ports `wgpu`'s Boids example, which is noteworthy because it involves both a compute and render pass.

https://github.com/tombh/boids

I found it very educational and wish I'd known sooner how simple it all was. Of course there's still some big holes in my knowledge, like there's some kind of undocumented convention for the ordering of arguments to GL entry points?

So I don't have a specific recommendation, I'm just a keen potential user of the project trying to navigate it in its current state. Maybe I'm still just a bit too early?

**Comment by AidanConnelly** [comment]
Hi Tom!

I'm also fairly new to both Rust and GPU programming, you should join the discord for this project if you haven't already.

Your Boids also seems like it might be worth submitting to [the "Share Your Creations" page on the GitHub 'discussion'](https://github.com/EmbarkStudios/rust-gpu/discussions/517)

**Comment by khyperia** [comment]
Thanks for the example! As @AidanConnelly, github discussions would be a great place to share this.

As for the severely lacking documentation - unfortunately yeah, it is indeed severely lacking, the two of us have been pretty overwhelmed with other parts of the project and are struggling to find time for documentation. Hopefully we can get on that soon! Thanks for pointing out the pain points, too, that really helps us figure out how to structure new docs we write.

**Comment by tombh** [comment]
@AidanConnelly thanks for the link!

@khyperia thanks for the reply! I don't think I explained myself well. Where I've got to is that I think this project is *more* ready than it appears. I think my associations with a project that is as functional as this one are that there is usually more fanfare about that in the README and such. So for me there's been an interesting dissonance "reading" the project: on the one hand I find myself surprised by how I'm continuously pulled into it through its lack (yes lack!) of showstopping issues, but on the other hand, when things do get tricky, my immediate response is something like, "oh right yeah, I've never been initiated into SPIRV, of course I can't use this project". But the fact is that familiarity with SPIRV isn't a hard requirement, it's just a bonus.

Ok, so my conclusion I think is something like this: *I should have a light touch with this project, not because of my unfamiliarity with SPIRV, but rather simply because the project is still beta*. Now of course the beta state is made clear in the README, but the beta state isn't so clear when I manage to have so much success with my pet projects 😂 I mean just the sheer fact of sharing the very same `struct`s between CPU and GPU is incredible to me. Let alone all the other joys of clippy, tests, etc on the GPU.

So transitioning out of beta isn't just about the readiness of the code, it's also about public expectations right? Whereas I, someone uninitiated into SPIRV and the deeper side of GPUs, see something very ready, others have higher expectations of what would constitute ready in this space.

Ok so my final, final point is not a question of the documentation, it's just this question of how much a part of the target audience I am?

Either way, thanks so much for making all your hard work public like this. It's an inspiration ✨
